app/controllers/ChatController.scala

```diff
@@ -34,33 +35,36 @@ class ChatController @Inject() (implicit val ec: ExecutionContext, cc: Controlle
               else Some(makeGetChatLink(chatId, page + 1, perPage, authenticatedRequest)),
               last = makeGetChatLink(chatId, lastPage, perPage, authenticatedRequest)))))
           Ok(chat ++ metadata)
-        case Left(`chatNotFound`) => BadRequest(chatNotFound)
+        case Left(`chatNotFound`) => NotFound(chatNotFound)
         case Left(_) => InternalServerError(internalError)
       }
 
   }
 
-  def getChats(mailbox: Mailbox, page: Page, perPage: PerPage): Action[AnyContent] = authenticatedUserAction.async {
+  def getChats(mailbox: Mailbox, page: Page, perPage: PerPage,
+    sort: Sort): Action[AnyContent] = authenticatedUserAction.async {
     authenticatedRequest =>
 
-      chatService.getChats(mailbox, page, perPage, authenticatedRequest.userId)
-        .map {
-          case Some((chatsPreviewDTO, totalCount, lastPage)) =>
-            val chats = Json.obj("chats" -> Json.toJson(chatsPreviewDTO))
-
-            val metadata = Json.obj("_metadata" -> Json.toJsObject(PaginationDTO(
-              totalCount,
-              PageLinksDTO(
-                self = makeGetChatsLink(mailbox, page, perPage, authenticatedRequest),
-                first = makeGetChatsLink(mailbox, Page(0), perPage, authenticatedRequest),
-                previous = if (page == 0) None
-                else Some(makeGetChatsLink(mailbox, page - 1, perPage, authenticatedRequest)),
-                next = if (page >= lastPage) None
-                else Some(makeGetChatsLink(mailbox, page + 1, perPage, authenticatedRequest)),
-                last = makeGetChatsLink(mailbox, lastPage, perPage, authenticatedRequest)))))
-            Ok(chats ++ metadata)
-          case None => InternalServerError(internalError)
-        }
+      if (List(SORT_BY_DATE, DEFAULT_SORT) contains sort.sortBy)
```

Comment:
You don't need to create a List just to check this. Use == with || or use pattern matching.